source/common/http/conn_manager_impl.cc

```diff
@@ -188,8 +188,9 @@ void ConnectionManagerImpl::checkForDeferredClose(bool skip_delay_close) {
     close = Network::ConnectionCloseType::FlushWrite;
   }
   if (drain_state_ == DrainState::Closing && streams_.empty() && !codec_->wantsToWrite()) {
-    doConnectionClose(close, absl::nullopt,
-                      StreamInfo::ResponseCodeDetails::get().DownstreamLocalDisconnect);
+    // We are closing a draining connection with no active streams and the codec has
+    // nothing to write. As such the details are unnecessary.
+    doConnectionClose(close, absl::nullopt, "deferred_close_on_drained_connection");
```

Comment:
I think it makes sense to crate a singleton with constants for these reasons, since it will b easier to see all possible reason strings.

Reply:
This list will be non-complete as close can dependent on dependencies i.e. QUICHE  which is why I originally didn't opt for this.  Happy to do as a follow up if you feel strongly about this.